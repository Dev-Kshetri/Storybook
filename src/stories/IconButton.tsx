import { FC } from "react";
import { IconButton as IconButtonComponent } from "./IconButton.styles";

export interface IconButtonProps {
    icon: JSX.Element;
    backgroundColor: string;
    variant: "default" | "outlined" | "filled";
    edgesColor: string;
}

export const IconButton = ({
    icon,
    backgroundColor,
    edgesColor,
    variant = "default",
    ...props
}: IconButtonProps) => {


    return (
        <IconButtonComponent
            backgroundColor={backgroundColor}
            edgesColor={edgesColor}
            variant={variant}
            className={variant}
        >
            {icon}
        </IconButtonComponent>
    )
}
