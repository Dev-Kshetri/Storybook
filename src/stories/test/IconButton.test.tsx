import { render } from "@testing-library/react";
import { IconButton } from "../IconButton";


describe('<IconButton> Test', () => {

    it('should match snapshot and have given classes', () => {

        //IconButton Default
        const { container: componentIconButton1, getByRole } = render(<IconButton
            icon={<span>üêí</span>}
            backgroundColor=""
            edgesColor=""
            variant="default"
        />);
        expect(componentIconButton1).toMatchSnapshot();
        expect(componentIconButton1.getElementsByClassName("default").length).toBe(1);
        expect("span").toBeInTheDocument();


        //IconButton Filled
        const { container: componentIconButton2 } = render(<IconButton
            icon={<span>üêí</span>}
            backgroundColor=""
            edgesColor=""
            variant="filled"
        />);
        expect(componentIconButton2).toMatchSnapshot();
        expect(componentIconButton2.getElementsByClassName("filled").length).toBe(1);

        //IconButton Outlined
        const { container: componentIconButton3 } = render(<IconButton
            icon={<span>üêí</span>}
            backgroundColor=""
            edgesColor=""
            variant="outlined"
        />);
        expect(componentIconButton3).toMatchSnapshot();
        expect(componentIconButton3.getElementsByClassName("outlined").length).toBe(1);

    })

})