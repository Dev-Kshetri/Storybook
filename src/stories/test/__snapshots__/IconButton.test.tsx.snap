// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton> Test should match snapshot and have given classes 1`] = `
<div>
  <button
    backgroundcolor=""
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium sc-bczRLJ eDWNr default css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    edgescolor=""
    tabindex="0"
    type="button"
    variant="default"
  >
    <span>
      🐒
    </span>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`<IconButton> Test should match snapshot and have given classes 2`] = `
<div>
  <button
    backgroundcolor=""
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium sc-bczRLJ eDWNr filled css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    edgescolor=""
    tabindex="0"
    type="button"
    variant="filled"
  >
    <span>
      🐒
    </span>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`<IconButton> Test should match snapshot and have given classes 3`] = `
<div>
  <button
    backgroundcolor=""
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium sc-bczRLJ eDWNr outlined css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    edgescolor=""
    tabindex="0"
    type="button"
    variant="outlined"
  >
    <span>
      🐒
    </span>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
